make download easier

earringyyr opened the pull request:
Show the download_file on the problem page if needed and show the data_file if the user is SuperUser.
What's more, hide the url in the problem_manege page if it leads to 404 page.
In passing, support that SuperUser can download the data_file by url: /download?type=data&id=problem_id

earringyyr committed:
feat(web):make download easier

Show the download_file on the problem page if needed and show the data_file if the user is SuperUser.
What's more, hide the url in the problem_manege page if it leads to 404 page.
In passing, support that SuperUser can download the data_file by url: /download?type=data&id=problem_id

billchenchina commented on web/app/controllers/download.php:
```diff
@@ -30,6 +30,14 @@
 			$file_name = "/opt/uoj/judger/uoj_judger/include/testlib.h";
 			$download_name = "testlib.h";
 			break;
+		case 'data':
```
https://github.com/BIDGroup/UOJ-Edu/blob/master/app/controllers/download.php#L33

billchenchina reviewed (commented):
Well...
https://github.com/BIDGroup/UOJ-Edu/blob/master/app/controllers/download.php#L33
估计只是碰巧了😂毕竟谁写都会这么写

billchenchina reviewed (approved):
LGTM

AdemJensen committed:
Merge branch 'master' into master

AdemJensen committed:
Replaced some uxexpected spaces.

Replaced some of the unexpected spaces in the code that makes the code a little tough to read.

AdemJensen committed:
Replaced some unexpected spaces with tab

The same issue as the problem.php, resolved already.

AdemJensen reviewed (approved):
LGTM+1
